### PR TITLE
compatibility enhancements: reconnect support, autogenerate devs

### PR DESCRIPTION
**Pull request by ffromani**
enhancements for better compatibility with the u/s official sample device plugin:
- trivial reconnect support, just the bare minimum we need to handle the most basic scenarios for testing
- autogenerate, if asked to, u/s compatible trivial devices. Note the much more powerful configuration file support is fuly kept and still the preferred option

### CODE REVIEW

**Comment by ffromani** [comment]
Note that even if we merge this PR, the plugin needs to create the `/tmp/Dev-$X` devices. So it may still be not fully compatible with u/s in generated mode (need to address that)

File: pkg/deviceconfig/deviceconfig.go [hunk 1]
```diff
@@ -59,7 +59,23 @@ func Parse(path, resName string) (*NodesDevices, error) {
 	if err != nil {
 		return nil, err
 	}
-	return conf, nil
+	return &conf, nil
+}
+
+func Generate(count int) *NodesDevices {
```
Comment (swatisehgal) [inline pkg/deviceconfig/deviceconfig.go:65]: I would suggest adding tests for this.

**Review by swatisehgal — COMMENTED** [review]
Looks good.

Left a comment about adding tests for the newly introduced `Generate` function .

File: pkg/deviceconfig/deviceconfig.go (hunk 1 above)
Comment (ffromani) [reply pkg/deviceconfig/deviceconfig.go:65]: done, thanks for pointing this out

**Review by swatisehgal — APPROVED** [review]
/lgtm

Thanks for making the updates.

**Pull request merged**